### PR TITLE
feat: update test suite location

### PR DESCRIPTION
**Pull request by kpodsiad**
needs https://github.com/scalameta/metals-vscode/pull/1043
VS Code updates range of test suite using their own heuristic but sometimes it yields wrong results. This should be handy also for potential non-vscode clients.

Todo: 
- [x] docs
- [ ] test
- [x] ~now I realized that VS Code doesn't need location, range is enough~ Actually, it doesn't matter that much.

I want to do some cleanup in Test Explorer, get rid of ballast code but this will happen on subsequent PRs.

### CODE REVIEW

**Comment by zmerr** [comment]
Oh, you’re back ❤️👻

File: metals/src/main/scala/scala/meta/internal/metals/testProvider/TestSuitesProvider.scala [hunk 1]
```diff
@@ -124,6 +137,34 @@ final class TestSuitesProvider(
...
+      if loc != entry.testClass.location
+    } yield {
+      val event = UpdateSuiteLocation(
+        entry.suiteInfo.fullyQualifiedName.value,
+        entry.suiteInfo.className.value,
+        loc
+      )
+      (entry.buildTarget, event)
+    }
+    events.groupBy(_._1).toList.map { case (buildTarget, events) =>
```
Comment (tgodzik) [inline metals/src/main/scala/scala/meta/internal/metals/testProvider/TestSuitesProvider.scala:160]:
```suggestion
    events.groupBy{case (target, _) }.toList.map { case (buildTarget, events) =>
```

File: metals/src/main/scala/scala/meta/internal/metals/testProvider/TestSuitesProvider.scala [hunk 2]
```diff
@@ -124,6 +137,34 @@ final class TestSuitesProvider(
...
+      BuildTargetUpdate(
+        buildTarget,
+        events.map(_._2)
```
Comment (tgodzik) [inline metals/src/main/scala/scala/meta/internal/metals/testProvider/TestSuitesProvider.scala:163]:
```suggestion
        events.map{case (_, event) => event}
```

File: metals/src/main/scala/scala/meta/internal/metals/testProvider/TestSuitesProvider.scala [hunk 3]
```diff
@@ -62,19 +63,31 @@ final class TestSuitesProvider(
...
+            getTestSuitesLocationUpdates(file, doc)
+          }
+        else Nil
+
+      val buildTargetUpdates =
+        if (index.hasTestCasesGranularity(file))
+          getTestCasesForPath(file, doc)
+        else Nil
+
+      updateClientIfNonEmpty(suiteLocationChanged ::: buildTargetUpdates)
```
Comment (tgodzik) [inline metals/src/main/scala/scala/meta/internal/metals/testProvider/TestSuitesProvider.scala:90]:
```suggestion
  docs.documents.headOption.foreach {
      case doc if isEnabled =>

      val suiteLocationChanged =
        if (index.contains(file))
          doc.toList.flatMap { doc =>
            getTestSuitesLocationUpdates(file, doc)
          }
        else Nil

      val buildTargetUpdates =
        if (index.hasTestCasesGranularity(file))
          getTestCasesForPath(file, doc)
        else Nil

      updateClientIfNonEmpty(suiteLocationChanged ::: buildTargetUpdates)
      case _ =>
```

**Review by tgodzik — COMMENTED** [review]
Some minor comments, but looks good!

**Review by tgodzik — APPROVED** [review]
LGTM!

**Pull request merged**